examples/sod/ArborX_DetailsSOD.hpp

```diff
@@ -0,0 +1,388 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILSSOD_HPP
+#define ARBORX_DETAILSSOD_HPP
+
+#include <ArborX.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+
+KOKKOS_INLINE_FUNCTION
+float rDelta(float r_min, float r_max, int num_sod_bins)
+{
+  return log10(r_max / r_min) / (num_sod_bins - 1);
+}
+
+KOKKOS_INLINE_FUNCTION
+int binID(float r_min, float r_max, float r, int num_sod_bins)
+{
+
+  float r_delta = rDelta(r_min, r_max, num_sod_bins);
+
+  int bin_id = 0;
+  if (r > r_min)
+    bin_id = (int)floor(log10(r / r_min) / r_delta) + 1;
+  if (bin_id >= num_sod_bins)
+    bin_id = num_sod_bins - 1;
+
+  return bin_id;
+}
```

Comment:
The number of bins and their treatment is coming from Cosmotools expectations. There is not really any flexibility on our side. There's already a bin for underflow (all particles `< r_min`), if that's what you mean. There's no bin for overflow, because that's not what CosmoTools expects. And we can't really ask them to change, because they will maintain their original version alongside ArborX.

Unit tests will wait.